test/e2e_node/mirror_pod_readiness_test.go

```diff
@@ -0,0 +1,135 @@
+	f := framework.NewDefaultFramework("mirror-pod")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+	ginkgo.Context("when create a mirror pod check readiness ", func() {
+		var ns, podPath, staticPodName, mirrorPodName string
+		ginkgo.BeforeEach(func(ctx context.Context) {
+			ns = f.Namespace.Name
+			staticPodName = "static-pod-" + string(uuid.NewUUID())
+			mirrorPodName = staticPodName + "-" + framework.TestContext.NodeName
+
+			podPath = framework.TestContext.KubeletConfig.StaticPodPath
+
+			ginkgo.By("create the static pod")
+			err := createStaticPodWithReadiness(podPath, staticPodName, ns,
+				imageutils.GetE2EImage(imageutils.Perl), v1.RestartPolicyAlways)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("wait for the mirror pod to be running")
+			gomega.Eventually(ctx, func(ctx context.Context) error {
+				return checkMirrorPodRunning(ctx, f.ClientSet, mirrorPodName, ns)
+			}, 2*time.Minute, time.Second*4).Should(gomega.BeNil())
+
+			ginkgo.By("wait for pod to be ready")
+			gomega.Eventually(ctx, func(ctx context.Context) error {
+				pod, err := f.ClientSet.CoreV1().Pods(ns).Get(ctx, mirrorPodName, metav1.GetOptions{})
+				if err != nil {
+					return fmt.Errorf("expected the mirror pod %q to be present: %v", mirrorPodName, err)
+				}
+				for _, c := range pod.Status.Conditions {
+					if c.Type == v1.PodReady && c.Status == v1.ConditionTrue {
+						return nil
+					}
+				}
+				return errors.New("pod needs to be ready")
+			}, 2*time.Minute, time.Second*4).Should(gomega.BeNil())
+		})
+		/*
+			Testname: Mirror Pod, readiness
+			Description: checks the readiness of a pod on termination
+		*/
+		ginkgo.It("should be unready with readiness probe failure on termination [NodeConformance]", func(ctx context.Context) {
+			// remove the static pod, pod should go unready
+			err := deleteStaticPod(podPath, staticPodName, ns)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("wait for the mirror pod to be not ready")
+			gomega.Eventually(ctx, func(ctx context.Context) error {
+				pod, err := f.ClientSet.CoreV1().Pods(ns).Get(ctx, mirrorPodName, metav1.GetOptions{})
+				if err != nil {
+					return fmt.Errorf("expected the mirror pod %q to be present: %v", mirrorPodName, err)
+				}
+				for _, c := range pod.Status.Conditions {
+					if c.Type == v1.PodReady && c.Status == v1.ConditionFalse {
+						return nil
+					}
+				}
+				return errors.New("pod needs to be unready")
+			}, 2*time.Minute, time.Second*4).Should(gomega.BeNil())
+		})
+	})
+})
+
+func createStaticPodWithReadiness(dir, name, namespace, image string, restart v1.RestartPolicy) error {
+	template := `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  containers:
+  - name: test
+    image: %s
+    restartPolicy: %s
+    command:
+      - bash
+      - -c
+      - "_term() { rm -f /tmp/ready ; sleep 10; exit 0; } ; trap _term SIGTERM; touch /tmp/ready ; while true; do echo \"hello\"; sleep 10; done"
```

Comment:
It would be very nice to have both the sleep shutdown and the grace period parameterizable - we need to test grace period of shutdown as well for static pods and this would be a great starting point.

It would be nice to verify that the transition from ready / not ready was detected and attributed to the new pod too?